joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -95,7 +94,7 @@ trait AstForDeclarationsCreator {
             astForFunctionDeclaration(nodeInfo, shouldCreateFunctionReference = true, shouldCreateAssignmentCall = true)
           case _ => astForNode(d)
         }
-        val defaultName = ast.root.collect { case r: HasName => r.name }
+        val defaultName = nameForFunctionDeclaration(ast)
```

Comment:
Could you not just do here what you do in `nameForFunctionDeclaration` and everything else would not need to change?